lib/web_scraper.dart

```diff
@@ -1,167 +1,220 @@
-  List<Map<String, dynamic>> getElement(String address, List<String> attribs) {
-    // Attribs are the list of attributes required to extract from the html tag(s) ex. ['href', 'title'].
-    if (_response == null) {
-      throw WebScraperException(
-          'getElement cannot be called before loadWebPage');
-    }
-    // Using html parser and query selector to get a list of particular element.
-    var document = parse(_response.body);
-    var elements = document.querySelectorAll(address);
-    // ignore: omit_local_variable_types
-    List<Map<String, dynamic>> elementData = [];
-
-    for (var element in elements) {
-      var attribData = <String, dynamic>{};
-      for (var attrib in attribs) {
-        attribData[attrib] = element.attributes[attrib];
-      }
-      elementData.add({
-        'title': element.text,
-        'attributes': attribData,
-      });
-    }
-    return elementData;
-  }
-}
-
-/// WebScraperException throws exception with specified message.
-class WebScraperException implements Exception {
-  var _message;
-  WebScraperException(String message) {
-    _message = message;
-  }
-  String errorMessage() {
-    return _message;
-  }
-}
+/*
+  Developed by Tushar Ojha
+  GitHub: https://github.com/tusharojha
+  Twitter: https://twitter.com/tusharojha
+  Feel free to improve the web_scraper library.
+*/
+
+library web_scraper;
+
+import 'dart:async';
+
+import 'package:html/parser.dart'; // Contains HTML parsers to generate a Document object.
+import 'package:http/http.dart'; // Contains a client for making API calls.
+
+import 'validation.dart';
+
+/// WebScraper Main Class.
+class WebScraper {
+  // Response Object of web scrapping the website.
+  var _response;
+
+  // Parsed document from the response inside the try/catch of the loadWebPage() method.
+  var document;
+
+  // Time elapsed in loading in milliseconds.
+  int timeElaspsed;
+
+  // Base url of the website to be scrapped.
+  String baseUrl;
+
+  Map<int, Client> Clients;
+
+  /// Creates the web scraper instance.
+  WebScraper(String baseUrl) {
+    var v = Validation().isBaseURL(baseUrl);
+    if (!v.isCorrect) {
+      throw WebScraperException(v.description);
+    }
+    this.baseUrl = baseUrl;
+  }
+
+  /// Loads the webpage into response object.
+  Future<bool> loadWebPage(String route) async {
+    if (baseUrl != null || baseUrl != '') {
+      final stopwatch = Stopwatch()..start();
+      var client = Client();
+
+      try {
+        _response = await client.get(baseUrl + route);
+        // Calculating Time Elapsed using timer from dart:core.
+        if (_response != null) {
+          timeElaspsed = stopwatch.elapsed.inMilliseconds;
+          stopwatch.stop();
+          stopwatch.reset();
+          // Parses the response body once it's retrieved and stores it on the var document defined on line 23.
+          document = parse(_response.body);

```

Comment:
Update the comment as we don't need to specify the line number where the variable is declared since we can navigate to that using vs code 👍

Reply:
Removed!